python/ql/src/Variables/ShadowBuiltin.ql

```diff
@@ -43,10 +44,11 @@ predicate white_list(string name) {
 }
 
 predicate shadows(Name d, string name, Scope scope, int line) {
-    exists(LocalVariable l | d.defines(l) and scope instanceof Function and
+    exists(LocalVariable l | d.defines(l) and
            l.getId() = name and
-           exists(Object::builtin(l.getId()))
+           exists(Builtin::builtin(l.getId()))
     ) and
+    scope instanceof Function and
```

Comment:
I'm starting to wonder if the restriction of scope to `Function` makes sense?

Looking at the original code 

```
predicate shadows(Name d, string name, Scope scope, int line) {
    exists(LocalVariable l | d.defines(l) and scope instanceof Function and
           l.getId() = name and
           exists(Object::builtin(l.getId()))
    ) and
    d.getScope() = scope and
    d.getLocation().getStartLine() = line and
    not white_list(name) and
    not optimizing_parameter(d)
}
```

it might simply have been an oversigt? What do you think?